### PR TITLE
Circleci project setup

### PR DESCRIPTION
**Pull request by BryceLTaylor**
### Resolves:

Makes progress to moving from Travis to CirlceCI for our CI.

### Changes:

Adds a circleCI configuration that will build and deploy for develop, hotfix, and release branches.  Will not deploy production.

### CODE REVIEW

**Comment by BryceLTaylor** [comment]
The build job is failing, most likely because it's running out of memory.  We need to either use less memory or get more of it before we can make the switch.

**Comment by BryceLTaylor** [comment]
Memory issues resolved.  Also switched to using `npm install` instead of `npm ci` because ci was causing the build to fail when dependencies used `BundledDependencies`.  Later we can look into switching back to npm ci.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+  build-test-deploy:
+    jobs:
+      - build-staging:
+          context:
+            - scratch-www-all
+            - scratch-www-staging
+          filters:
+            branches:
+              only:
+                - circleci-project-setup # remove this before releasing
```
Comment (rschamp) [inline .circleci/config.yml:107]: We should do this before merging.

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                - develop
+                - /^hotfix\/.*/
+                - /^release\/.*/
```
Comment (rschamp) [inline .circleci/config.yml:110]: We can check this syntax by pushing this branch to a hotfix/ and/or release/ branch name.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+      - deploy-staging:
+          context:
+            - scratch-www-all
+            - scratch-www-staging
+          requires:
+            - build-staging
+          filters:
+            branches:
+              only:
+                - circleci-project-setup # remove this before releasing
```
Comment (rschamp) [inline .circleci/config.yml:120]: We should do this before merging.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                - develop
+                - /^hotfix\/.*/ # make sure this syntax works.  may need a regex
+                - /^release\/.*/
```
Comment (rschamp) [inline .circleci/config.yml:123]: Check syntax as above

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            - scratch-www-staging
+          requires:
+            - deploy-staging
+          filters:
+            branches:
+              only:
+                - circleci-project-setup # remove this before releasing
+                - develop
+                - /^hotfix\/.*/
+                - /^release\/.*/\
```
Comment (rschamp) [inline .circleci/config.yml:136]: Check syntax

File: .circleci/config.yml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+      keys:
+        - v1-build-{{ .Revision }}
+        - v1-build-
+  - &build
+    <<: *defaults
+    resource_class: medium+
+    steps:
+      - *restore_git_cache
+      - checkout
+      # - *restore_npm_cache
```
Comment (rschamp) [inline .circleci/config.yml:48]: Now that we're using install I think this becomes useful again.

**Review by rschamp — CHANGES_REQUESTED** [review]
Just some cleanup needed before merging. Nice work!

File: .circleci/config.yml (hunk 2 above)
Comment (BryceLTaylor) [reply .circleci/config.yml:109]: Tested this out, both hotfix and release branches run the whole workflow

File: .circleci/config.yml (hunk 4 above)
Comment (BryceLTaylor) [reply .circleci/config.yml:121]: Tested this out, both hotfix and release branches run the whole workflow

File: .circleci/config.yml (hunk 5 above)
Comment (BryceLTaylor) [reply .circleci/config.yml]: Tested this out, both hotfix and release branches run the whole workflow

File: .circleci/config.yml (hunk 1 above)
Comment (BryceLTaylor) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 6 above)
Comment (BryceLTaylor) [reply .circleci/config.yml]: done

File: .circleci/config.yml (hunk 3 above)
Comment (BryceLTaylor) [reply .circleci/config.yml]: removed

**Review by rschamp — APPROVED** [review]

**Pull request merged**